hydroflow/src/compiled/pull/symmetric_hash_join.rs

```diff
@@ -44,39 +42,39 @@ where
 
     fn next(&mut self) -> Option<Self::Item> {
         loop {
-            if let Some((k, v, vs)) = &mut self.state.lbuffer {
+            if let Some((k, v, vs)) = &mut self.state.0.buffer {
                 // TODO(justin): unnecessary clone (sometimes).
                 let result = (k.clone(), (v.clone(), vs.pop().unwrap()));
                 if vs.is_empty() {
-                    self.state.lbuffer = None;
+                    self.state.0.buffer = None;
                 }
                 return Some(result);
-            } else if let Some((k, v, vs)) = &mut self.state.rbuffer {
+            } else if let Some((k, v, vs)) = &mut self.state.1.buffer {
                 // TODO(justin): unnecessary clone (sometimes).
                 let result = (k.clone(), (vs.pop().unwrap(), v.clone()));
                 if vs.is_empty() {
-                    self.state.rbuffer = None;
+                    self.state.1.buffer = None;
                 }
                 return Some(result);
             }
 
             if let Some((k, v1)) = self.lhs.next() {
```

Comment:
could be DRYer -- both left and right use the same code, just different arguments. Factor out the logic?